src/linear_algebra/finsupp.lean

```diff
@@ -550,6 +550,11 @@ protected def dom_lcongr {α₁ α₂ : Type*} (e : α₁ ≃ α₂) :
   (α₁ →₀ M) ≃ₗ[R] (α₂ →₀ M) :=
 (finsupp.dom_congr e : (α₁ →₀ M) ≃+ (α₂ →₀ M)).to_linear_equiv (lmap_domain M R e).map_smul
 
+lemma dom_lcongr_apply {α₁ : Type*} {α₂ : Type*} (e : α₁ ≃ α₂) (v : α₁ →₀ M) (i : α₂) :
+  (finsupp.dom_lcongr e : _ ≃ₗ[R] _) v i = v (e.symm i) :=
+by { conv_lhs { rw ← e.apply_symm_apply i },
+     exact finsupp.map_domain_apply e.injective v (e.symm i) }
```

Comment:
Here's the existing proof:
```suggestion
lemma dom_lcongr_apply {α₁ : Type*} {α₂ : Type*} (e : α₁ ≃ α₂) (v : α₁ →₀ M) (i : α₂) :
  (finsupp.dom_lcongr e : _ ≃ₗ[R] _) v i = v (e.symm i) :=
map_domain_equiv_apply v i
```
I'd probably lean towards the following statement though, as `simp` can take things from here:
```
@[simp]
lemma dom_lcongr_apply {α₁ : Type*} {α₂ : Type*} (e : α₁ ≃ α₂) (v : α₁ →₀ M) (i : α₂) :
  ⇑(finsupp.dom_lcongr e : (α₁ →₀ M) ≃ₗ[R] _) = finsupp.dom_congr e :=
rfl
```
As a nice bonus, this provides a lemma to convert `dom_congr` in a goal into `dom_lcongr`.

Reply:
(Especially because I, as a user of this lemma, really want to see `v (e.symm i)` and not `finsupp.dom_congr` in my goal.)

Reply:
Based on that argument, the RHS of `finsupp.dom_congr_apply` is also wrong.

The idea behind simplifying to `map_domain` is that you can then state all the interesting lemmas about `map_domain` just once, instead of having to state them for every variant of `map_domain` (`map_domain.add_monoid_hom`, `dom_congr`, `dom_lcongr`, etc).

Conversely, if I have `dom_congr` in my goal and want to rewrite it into `dom_lcongr`, it seems more intuitive to me to be able to rewrite along a lemma that unfolds `dom_lcongr` one level rather than fully unfold to `v (e.symm i)` and then re-fold.

Reply:
Coming back to it, I don't care so much about this point. So I went ahead and applied your suggestion.

Reply:
Thanks!